pkg/datasources/schemas.go

```diff
@@ -48,11 +49,20 @@ func Schemas() *schema.Resource {
 
 func ReadSchemas(d *schema.ResourceData, meta interface{}) error {
 	db := meta.(*sql.DB)
+	client := sdk.NewClientFromDB(db)
+	ctx := context.Background()
 	databaseName := d.Get("database").(string)
+	databaseID := sdk.NewAccountObjectIdentifier(databaseName)
 
 	log.Printf("[DEBUG] database name %s", databaseName)
 
-	currentSchemas, err := snowflake.ListSchemas(databaseName, db)
+	currentSchemas, err := client.Schemas.Show(ctx, &sdk.ShowSchemaOptions{
+		In: &sdk.SchemaIn{
+			Database: sdk.Bool(true),
+			Name:     databaseID,
+		},
+	})
+
 	if errors.Is(err, sql.ErrNoRows) {
```

Comment:
I guess that now this won't be the same error.

For other datasources we have moved to plan `err != nil` when migrating to SDK (e.g. check `database_roles.go`).